hie-plugin-api/Haskell/Ide/Engine/PluginDescriptor.hs

```diff
@@ -214,7 +233,8 @@ data ParamVal (t :: ParamType) where
 class ValidResponse a where
   jsWrite :: a -> Object -- ^ Serialize to JSON Object
   jsRead  :: Object -> Parser a -- ^ Read from JSON Object
-
+  responseTypeName :: a -> T.Text -- ^ Type name in command description.
```

Comment:
Bringing in the discussion from #109, if the type is clearly serialised, will this be necessary?

In other words, do we need a tag for the command context as well as the type?

The one example is a HaRe Diff, which is currently supplied as just a list of files, but will be a full featured diff soon.

Is it important to know that it is a refactor diff rather than just a diff? Can this knowledge be accessed from the command that was sent, and hence the known reply? Should we tag it with the "plugin:command name" instead?

To me this comes back to the discussion about semantic types / widgets.